deployment/ccip/changeset/cs_active_candidate.go

```diff
@@ -17,37 +17,68 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
 )
 
+var (
+	_ deployment.ChangeSet[PromoteAllCandidatesChangesetConfig]  = PromoteAllCandidatesChangeset
+	_ deployment.ChangeSet[AddDonAndSetCandidateChangesetConfig] = SetCandidatePluginChangeset
+)
+
+type PromoteAllCandidatesChangesetConfig struct {
+	HomeChainSelector uint64
+	NewChainSelector  uint64
+	Nodes             deployment.Nodes
+}
+
+func (p PromoteAllCandidatesChangesetConfig) Validate() error {
```

Comment:
we can accept the environment to do a lot more validation here: 
- Ensure the nodeIDs passed exist in the offchainClient
- Ensure there is a candidate in CCIPHome